### PR TITLE
Madiha - JS Excercises

### PR DESCRIPTION
**Pull request by diya2214**
my work in class today

### CODE REVIEW

File: week-1-practise/level-2/4-currency-conversion-2.js [hunk 1]
```diff
@@ -7,8 +7,17 @@
...
-function convertToBRL(price) {}
+function convertToUSD(price) {
+  var conversionUSD = price * 1.4;
+  return addComission(conversionUSD);
+}
+function convertToBRL(price) {
+  var conversionBRL = price * 4.9;
+  return addComission(conversionBRL);
+}
+function addComission(conversion) {
```
Comment (mtbdata711) [inline week-1-practise/level-2/4-currency-conversion-2.js]: In awe at how clear and easy to understand your code is Madiha! Did you get the .toFixed to run fully?

Comment (diya2214) [reply week-1-practise/level-2/4-currency-conversion-2.js]: rhanks

File: week-1/K-functions-parameters/exercise.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Complete the function so that it takes input parameters
-function multiply() {
+function multiply(a, b) {
```
Comment (nbogie) [inline week-1/K-functions-parameters/exercise.js]: Great!

File: week-1/K-functions-parameters/exercise3.js [hunk 3]
```diff
@@ -1,5 +1,8 @@
 // Write your function here
```
Comment (nbogie) [inline week-1/K-functions-parameters/exercise3.js]: Yep, the function is right!  There's a spelling error in your English, though :)

File: week-1/K-functions-parameters/exercise4.js [hunk 4]
```diff
@@ -1,5 +1,7 @@
 // Declare your function first
```
Comment (nbogie) [inline week-1/K-functions-parameters/exercise4.js]: Perfect

File: week-1/K-functions-parameters/exercise5.js [hunk 5]
```diff
@@ -1,5 +1,7 @@
 // Declare your function here
-
+function createLongGreeting(name, age) {
```
Comment (nbogie) [inline week-1/K-functions-parameters/exercise5.js]: Great!  This is correct.  If I wanted to be REALLY unfair I could say that your greeting will be grammatically incorrect if the person's age is `1`.

File: week-1/L-functions-nested/excercise.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+function createMessage(message) {
```
Comment (nbogie) [inline week-1/L-functions-nested/excercise.js]: Great! Your program accomplishes what it needs to do, and the answers are correct.  
HOWEVER... given that "You should have one function that calculates the percentage, and one function that creates a message" - I would argue: 
1) the "creation" of the message should include the concatenation of its parts.
2) Your function to "calculate the percentage" does too much.  It doesn't only calculate a percentage, it calculates TWO percentages, creates and displays messages about them.
Can you change your program to make use of a function that only calculates a percentage based on the numbers given to it?

File: week-1/L-functions-nested/exercise2.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (nbogie) [inline week-1/L-functions-nested/exercise2.js]: Yep, great stuff!
I might have given a more specific name to the function which created a shouty greeting - but that's a very minor point.

File: week-1/L-functions-nested/excercise.js (hunk 6 above)
Comment (nbogie) [reply week-1/L-functions-nested/excercise.js]: A very minor point: watch out spelling 'exercise'.  It's not 'excercise' which you've used in a few places.

File: week-1-practise/level-1/1-syntax-error-1.js [hunk 8]
```diff
@@ -1,14 +1,16 @@
...
-
 // Answer these questions:
 // 1. How many parameters does the function take?
+// The above function takes three parameters.
 // 2. What is the function name?
-// 3. Where is the function called? with what parameters?
\ No newline at end of file
+// Functiin name is addNumbers
+// 3. Where is the function called? with what parameters?
+// Function is called on line 7 var result = addNumbers(1,3,4)
```
Comment (nbogie) [inline week-1-practise/level-1/1-syntax-error-1.js]: Good!
 // 3. Where is the function called? with what parameters?
The parameters passed to the function call are 1, 3, 4 (actually, these are more correctly called the arguments)

File: week-1-practise/level-1/2-syntax-error-2.js [hunk 9]
```diff
@@ -1,8 +1,8 @@
 // There is a syntax error in this code. Fix it.
 
-function addNumbers(a, b, c)
-return a + b + c;
-
+function addNumbers(a, b, c) {
+  return a + b + c;
+}
```
Comment (nbogie) [inline week-1-practise/level-1/2-syntax-error-2.js]: Fixed the indentation, too!  Great!

File: week-1-practise/level-1/8-function-output-1.js [hunk 10]
```diff
@@ -1,12 +1,17 @@
...
+// It calls a function to return a number by generatingrandon number between 0 and 1
+// and multiply it with 10.Then it logs the result .
 // - How many parameters does the function take?
+//  There is no parameters for this function
 // - What is the function name?
-// - Where is the function called? with what parameters?
\ No newline at end of file
+//   Function name is getNumber
+// - Where is the function called? with what parameters?
+//  Function is called on line 5 var result = getNumber()
```
Comment (nbogie) [inline week-1-practise/level-1/8-function-output-1.js]: Here I'd say: It's called on line 5 with no parameters.

Comment (nbogie) [reply week-1-practise/level-1/8-function-output-1.js]: Excellent!

File: week-1-practise/level-1/9-function-output-2.js [hunk 11]
```diff
@@ -1,14 +1,20 @@
 function s(w1, w2) {
-    return w1.concat(w2);
+  return w1.concat(w2);
 }
 
-var result = s('code', 'your');
+var result = s("code ", "your");
```
Comment (nbogie) [inline week-1-practise/level-1/9-function-output-2.js]: Actually, you can use " or ' quote marks.  Most important is to be consistent with your team mates (or yourself), or the existing code of whatever project you're working on.

File: week-1-practise/level-1/9-function-output-2.js [hunk 12]
```diff
@@ -1,14 +1,20 @@
...
 console.log(result);
 
 // Answer these questions:
 // - What does this program do? Run it and see, or Google some of the methods used ot understand them.
+//   It calls a function to concat 2 words passed as parameters (using a method concat())
```
Comment (nbogie) [inline week-1-practise/level-1/9-function-output-2.js]: Yes!  I'd call these words strings, but that's fine, and just to show that you know what concatenation is I'd use a different word.  e.g. "join".

File: week-1-practise/level-1/9-function-output-2.js [hunk 13]
```diff
@@ -1,14 +1,20 @@
...
 // - How many parameters does the function take?
+//   The function takes 2 parameters
 // - What is the function name?
+///  Function name is s
 // - Where is the function called? with what parameters?
+/// Function is called on line 5 and the parameters are code , your
```
Comment (nbogie) [inline week-1-practise/level-1/9-function-output-2.js]: Great.  To be more precise, the parameters are the strings "code" and "your".

File: week-1-practise/level-1/9-function-output-2.js [hunk 14]
```diff
@@ -1,14 +1,20 @@
...
 // - What does the function return?
-// - How can this function be improved?
\ No newline at end of file
+/// Its returns a string codeyour
+// - How can this function be improved?
+//// This function can be improved by adding a space at the end of first parameter 'code '
```
Comment (nbogie) [inline week-1-practise/level-1/9-function-output-2.js]: Excellent.

File: week-1-practise/level-2/1-tax.js [hunk 15]
```diff
@@ -5,7 +5,9 @@
   Sales tax is 20% of the price of the product
 */
 
-function calculateSalesTax() {}
+function calculateSalesTax(price) {
```
Comment (nbogie) [inline week-1-practise/level-2/1-tax.js]: yep good

File: week-1-practise/level-2/2-format-currency.js [hunk 16]
```diff
@@ -6,7 +6,9 @@
   Write a function that transforms numbers into the format £0.00
 */
 
-function formatCurrency() {}
+function formatCurrency(number) {
+  return "£" + number.toFixed(2);
+}
```
Comment (nbogie) [inline week-1-practise/level-2/2-format-currency.js]: yep

File: week-1-practise/level-2/3-currency-conversion.js [hunk 17]
```diff
@@ -5,7 +5,9 @@
   Write a function that converts a price to USD (exchange rate is 1.4 $ to £)
 */
 
-function convertToUSD() {}
+function convertToUSD(price) {
+  return price * 1.4;
+}
```
Comment (nbogie) [inline week-1-practise/level-2/3-currency-conversion.js]: yep

File: week-1-practise/level-2/4-currency-conversion-2.js [hunk 18]
```diff
@@ -7,8 +7,17 @@
...
+  return conversion + conversion.toFixed(2) * 0.01; // adding 1% comission to converted price
```
Comment (nbogie) [inline week-1-practise/level-2/4-currency-conversion-2.js]: Good, this works.  But there is something to be careful of here.  .toString() has actually returned a *string* here, and it is being converted back to a number for you.  
It is much better to keep things as numbers as long as possible, and call toFixed() on the final number!

File: week-1-practise/level-3/1-piping.js [hunk 19]
```diff
@@ -11,6 +11,7 @@
 */
 
 var startingValue = 5;
+result = format(add(multiply(startingValue, 2), 10));
```
Comment (nbogie) [inline week-1-practise/level-3/1-piping.js]: perfect

File: week-1-practise/level-1/1-syntax-error-1.js (hunk 8 above)
Comment (diya2214) [reply week-1-practise/level-1/1-syntax-error-1.js]: got it

File: week-1-practise/level-1/2-syntax-error-2.js (hunk 9 above)
Comment (diya2214) [reply week-1-practise/level-1/2-syntax-error-2.js]: thanks

File: week-1-practise/level-1/8-function-output-1.js (hunk 10 above)
Comment (diya2214) [reply week-1-practise/level-1/8-function-output-1.js]: will keep that in mind

File: week-1-practise/level-1/9-function-output-2.js (hunk 11 above)
Comment (diya2214) [reply week-1-practise/level-1/9-function-output-2.js]: ok

File: week-1-practise/level-1/9-function-output-2.js (hunk 12 above)
Comment (diya2214) [reply week-1-practise/level-1/9-function-output-2.js]: yes definitely a better word

File: week-1-practise/level-1/9-function-output-2.js (hunk 14 above)
Comment (diya2214) [reply week-1-practise/level-1/9-function-output-2.js]: thanks

File: week-1-practise/level-2/4-currency-conversion-2.js (hunk 18 above)
Comment (diya2214) [reply week-1-practise/level-2/4-currency-conversion-2.js]: will do that

File: week-1/E-strings/exercise.js [hunk 20]
```diff
@@ -1,3 +1,6 @@
 // Start by creating a variable `message`
 
+var message = 'This is a String';
```
[the remark this answers is not in the document]
Comment (diya2214) [reply week-1/E-strings/exercise.js]: thanks

File: week-1/G-strings-methods/exercise.js [hunk 21]
```diff
@@ -1,3 +1,5 @@
 // Start by creating a variable `message`
-
+var name = 'Madiha';
+nameLength = name.length;
```
[the remark this answers is not in the document]
Comment (diya2214) [reply week-1/G-strings-methods/exercise.js]: will be careful in future

File: week-1/G-strings-methods/exercise.js [hunk 22]
```diff
@@ -1,3 +1,5 @@
...
+var message = 'My name is ' + name + '. My name is ' + nameLength + ' character is long';
```
[the remark this answers is not in the document]
Comment (diya2214) [reply week-1/G-strings-methods/exercise.js]: fixed in new pull

File: week-1/G-strings-methods/exercise2.js [hunk 23]
```diff
@@ -1,3 +1,7 @@
 var name = " Daniel  ";
+nameLength = name.length;
```
[the remark this answers is not in the document]
Comment (diya2214) [reply week-1/G-strings-methods/exercise2.js]: fixed

File: week-1/H-numbers/exercise.js [hunk 24]
```diff
@@ -1 +1,6 @@
 // Start by creating a variables `numberOfStudents` and `numberOfMentors`
+var numberOfStudents = 25;
+var numberOfMentors = 15;
+console.log('Total number of Students= ' + numberOfStudents);
+console.log('Total Mentors = ' + numberOfMentors);
+console.log('Total number of students and mentors =' + numberOfStudents + numberOfMentors);
```
[the remark this answers is not in the document]
Comment (diya2214) [reply week-1/H-numbers/exercise.js]: its just joining 2 variables rather than adding .... fixed thanks

**Pull request closed without merging**